scipy/linalg/_basic.py

```diff
@@ -1014,6 +1019,11 @@ def det(a, overwrite_a=False, check_finite=True):
 
     """
     a1 = _asarray_validated(a, check_finite=check_finite)
+     
+    # handle empty array, determinate of empty array (identity array) is 1.
+    if a1.size == 0:
+        return 1.0
```

Comment:
An empty array is not identity array, this should return np.nan if empty array is square otherwise should error out. See `numpy.linalg.det(np.empty([2, 2]))`, numpy.linalg.det([]),  and `numpy.linalg.det(np.empty([]))` for different error messages

Reply:
Apologies I misspoke and should have written that the determinant of an empty matrix is the identity matrix.  I'm having a bit of trouble understanding how to best initialize an empty matrix. 

If i'm reading correctly, an empty matrix has at least the number of rows or columns being zero, ie dimensions of `[0,2]`, `[2,0]`, or `[0,0]` for example. In the case of `linalg.det()`, if the matrix passed must be square then the only option for an empty matrix would then be `[0,0]`.

```
>>> numpy.linalg.det(np.empty([]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 180, in det
  File "/usr/local/lib/python3.10/site-packages/numpy/linalg/linalg.py", line 2150, in det
    _assert_stacked_2d(a)
  File "/usr/local/lib/python3.10/site-packages/numpy/linalg/linalg.py", line 197, in _assert_stacked_2d
    raise LinAlgError('%d-dimensional array given. Array must be '
numpy.linalg.LinAlgError: 0-dimensional array given. Array must be at least two-dimensional
```
```
>>> numpy.linalg.det([])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 180, in det
  File "/usr/local/lib/python3.10/site-packages/numpy/linalg/linalg.py", line 2150, in det
    _assert_stacked_2d(a)
  File "/usr/local/lib/python3.10/site-packages/numpy/linalg/linalg.py", line 197, in _assert_stacked_2d
    raise LinAlgError('%d-dimensional array given. Array must be '
numpy.linalg.LinAlgError: 1-dimensional array given. Array must be at least two-dimensional
```
```
>>> numpy.linalg.det(np.empty([2,2]))
-4.887546495707343e+152
```
The results in the above snippet make me think that `[]`, `np.empty([])`, and `np.empty([2,2])` are matrices with dimensions that are non-zero.  In the linked issue, gh-17658, the initial post uses the following method of initializing am empty matrix 
```
>>> a = np.array([]).reshape((0,0))
```
which results in the determinant being 1. 
```
>>> np.linalg.det(a)
1.0
```
Based on reading I've done this seems to be correct for an empty matrix with the dimensions `[0,0]`.

Does the definition of the determinant of an empty matrix I'm using seem correct to you?

Reply:
My bad `numpy.linalg.det(np.empty([2,2]))` is not empty but filled with random numbers depending on what was on that piece of memory. I was meant to write (2, 0).